Moved startup options factory initialization to Configuration object

joao-r-reis opened the pull request:
This change removes the direct dependency between `Connection` and `StartupOptionsFactory`

jorgebay commented:
It looks like we need to add validation to `StartupRequest` constructor parameter.

joao-r-reis commented:
> 
> 
> It looks like we need to add validation to `StartupRequest` constructor parameter.

Yep, added the validation and removed the Mocked instances from integration tests that were causing the error.
